### PR TITLE
Add default values for runtime config

### PR DESCRIPTION
**Pull request by ijjk**
Makes sure default values for `publicRuntimeConfig` and `serverRuntimeConfig` are set. Also moves the error when `publicRuntimeConfig` is used with `target: 'serverless'` into the config loading. 

Fixes another aspect of #7713

### CODE REVIEW

**Review by Timer — APPROVED** [review]

**Review by Timer — CHANGES_REQUESTED** [review]
Can we test that these are defaulted for the case we're fixing?

**Comment by ijjk** [comment]
Yup, added a test for it

File: packages/next-server/server/config.ts [hunk 1]
```diff
@@ -39,6 +39,8 @@ const defaultConfig: { [key: string]: any } = {
     documentMiddleware: false,
     publicDirectory: false,
   },
+  serverRuntimeConfig: {},
+  publicRuntimeConfig: {},
```
Comment (timneutkens) [inline packages/next-server/server/config.ts]: This was not set intentionally as it will propagate into __NEXT_DATA__ when set

Comment (ijjk) [reply packages/next-server/server/config.ts]: It's updated to not propagate to `__NEXT_DATA__` if it's an empty object 👍

File: packages/next-server/lib/runtime-config.ts [hunk 2]
```diff
@@ -4,6 +4,11 @@ export default () => {
   return runtimeConfig
 }
 
-export function setConfig(configValue: any) {
-  runtimeConfig = configValue
+export function setConfig(
+  { publicRuntimeConfig, serverRuntimeConfig } = {} as any
+) {
+  runtimeConfig = {
+    publicRuntimeConfig: publicRuntimeConfig || {},
```
Comment (timneutkens) [inline packages/next-server/lib/runtime-config.ts]: When would you need to fall back to `{}` if there is literally no case where it's not set?

Comment (ijjk) [reply packages/next-server/lib/runtime-config.ts]: Per https://github.com/zeit/next.js/pull/7847#discussion_r301797505 `publicRuntimeConfig` isn't present when loading from `__NEXT_DATA__` if it's empty.

Comment (timneutkens) [reply packages/next-server/lib/runtime-config.ts]: But it doesn't have to be set either in that case

Comment (ijjk) [reply packages/next-server/lib/runtime-config.ts]: Updated to prevent the bundle size increase

**Review by Timer — CHANGES_REQUESTED** [review]
Can we get a test that makes sure these are missing from NEXT_DATA when empty?

**Comment by ijjk** [comment]
Yup added a test to make sure it's not added to `__NEXT_DATA__` when it's the default value

**Review by Timer — APPROVED** [review]
LGTM

**Pull request merged**